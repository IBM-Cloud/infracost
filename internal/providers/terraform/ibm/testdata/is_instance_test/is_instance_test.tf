
terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
      version = "~> 1.12.0"
    }
  }
}

provider "ibm" {
    region = "us-south"
}

resource "ibm_is_vpc" "testVpc" {
  name = "test-vpc"
}

resource "ibm_is_subnet" "testSubnet" {
  name            = "test-subnet"
  vpc             = ibm_is_vpc.testVpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
}

resource "ibm_is_ssh_key" "testSshKey" {
  name       = "test-ssh"
  public_key = file("~/.ssh/id_rsa.pub")
}


resource "ibm_is_instance" "testInstance" {
  name    = "test-instance-1"
  image   = "r006-f137ea64-0d27-4d81-afe0-353fd0557e81"
  profile = "cx2-2x4"

  primary_network_interface {
    subnet = ibm_is_subnet.testSubnet.id
  }

  network_interfaces {
    name   = "eth1"
    subnet = ibm_is_subnet.testSubnet.id
  }

  vpc  = ibm_is_vpc.testVpc.id
  zone = "us-south-1"
  keys = [ibm_is_ssh_key.testSshKey.id]
}
