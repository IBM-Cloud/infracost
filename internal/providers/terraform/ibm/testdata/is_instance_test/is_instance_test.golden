
 Name                                                        Monthly Qty  Unit                        Monthly Cost 
                                                                                                                   
 ibm_is_instance.testBalancedInstance                                                                              
 ├─ CPU hours (32 CPUs, us-east-1)                                23,360  CPU hours                        $558.58 
 ├─ Memory hours (128 GB, us-east-1)                              93,440  Memory hours                     $453.19 
 └─ Storage GB hours (600 GB * 2, us-east-1)                     876,000  Storage GB hours                  $96.00 
                                                                                                                   
 ibm_is_instance.testBalancedInstanceWithBootVolume                                                                
 ├─ CPU hours (8 CPUs, us-east-1)                                  5,840  CPU hours                        $139.64 
 ├─ Memory hours (32 GB, us-east-1)                               23,360  Memory hours                     $113.30 
 └─ Boot volume (boot-volume-label, 150 GB)                      109,500  GB Hours                          $12.05 
                                                                                                                   
 ibm_is_instance.testComputeInstance                                                                               
 ├─ CPU hours (2 CPUs, us-east-1)                                  1,460  CPU hours                         $34.91 
 └─ Memory hours (4 GB, us-east-1)                                 2,920  Memory hours                      $21.07 
                                                                                                                   
 ibm_is_instance.testGpuInstance                                                                                   
 ├─ CPU hours (16 CPUs, us-east-1)                                11,680  CPU hours                        $298.39 
 ├─ Memory hours (128 GB, us-east-1)                              93,440  Memory hours                     $355.88 
 └─ Gpu hours (2 GPUs, Tesla V100, us-east-1)                      1,460  Gpu hours                      $2,920.00 
                                                                                                                   
 ibm_is_instance.testIbmZInstance                                                                                  
 ├─ CPU hours (16 CPUs, us-east-1)                                11,680  CPU hours                        $688.75 
 └─ Memory hours (64 GB, us-east-1)                               46,720  Memory hours                     $529.49 
                                                                                                                   
 ibm_is_instance.testInstanceWithoutUsage                                                                          
 ├─ CPU hours (2 CPUs, us-east-1)                    Monthly cost depends on usage: $0.0239116112 per CPU hours    
 └─ Memory hours (4 GB, us-east-1)                   Monthly cost depends on usage: $0.0072168965 per Memory hours 
                                                                                                                   
 ibm_is_vpc.testVpc                                                                                                
 ├─ VPC instance                                                       1  Instance                           $0.00 
 ├─ VPC egress free allowance (first 5GB)            Monthly cost depends on usage: $0.00 per GB                   
 └─ VPC egress us-east (first 9995 GB)               Monthly cost depends on usage: $0.087 per GB                  
 └─ VPC egress us-east (next 40000 GB)               Monthly cost depends on usage: $0.083 per GB                  
 └─ VPC egress us-east (next 100000 GB)              Monthly cost depends on usage: $0.07 per GB                   
 └─ VPC egress us-east (over 149995 GB)              Monthly cost depends on usage: $0.05 per GB                   
                                                                                                                   
 OVERALL TOTAL                                                                                           $6,221.25 
──────────────────────────────────
9 cloud resources were detected:
∙ 7 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 2 were free:
  ∙ 1 x ibm_is_ssh_key
  ∙ 1 x ibm_is_subnet